src/cdk/table/table.ts

```diff
@@ -1085,7 +1085,10 @@ export class CdkTable<T>
    * re-render that section.
    */
   private _renderUpdatedColumns(): boolean {
-    const columnsDiffReducer = (acc: boolean, def: BaseRowDef) => acc || !!def.getColumnsDiff();
+    const columnsDiffReducer = (acc: boolean, def: BaseRowDef) => {
+      const diff = !!def.getColumnsDiff();
```

Comment:
Should we leave a comment here with an explanation?

Reply:
Ah right thanks for the reminder - added